onnxscript/tests/function_libs/torch_lib/ops_test_common.py

```diff
@@ -581,11 +596,15 @@ def normal_xfail_skip_test_behaviors(
         yield
     # We could use `except (AssertionError, RuntimeError, ...) as e:`, but it needs
     # to go over all test cases to find the right exception type.
-    except Exception as e:  # pylint: disable=broad-exception-caught
+    except Exception:  # pylint: disable=broad-exception-caught
         if test_behavior is None:
-            raise e
+            raise
         if test_behavior == "xfail":
-            pytest.xfail(reason=reason)
+            if raises is not None:
+                with pytest.raises(raises):
```

Comment:
Passing in the unittest instance, can we do

```python
        if test_behavior == "xfail":
            if raises is not None:
                with test_case.assertRaisesRegex(raises, expected_error_message_regex):
                    raise
            pytest.xfail(reason=reason)
```

Reply:
nvm, I think we need a context manager similar to this but wrapped over subtest calling.

Reply:
It should work for any subtests, but I am not sure how to support the test methods?